### ISSUE TITLE
[a11y]: `SidePanel` stories need to all use `launcherButtonRef` prop 

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#5766]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

`2.45.0-rc.1`

### Automated testing tool and ruleset

Manual

### Assistive technology

JAWS

### Description

Focus should be returned to the button that opened the side panel. This should be just updates to `SidePanel.stories.jsx` but are required for screen reader experience.

### WCAG 2.1 Violation

4.1.2 Name Role Value

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic

### Steps to reproduce

- [Visit side panel story](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic)
- Open side panel
- Close side panel
- Focus is not returned to the button that opened that panel

Behavior should match [this story](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--return-focus-to-open-button&globals=viewport:basic).

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
